README.md

```diff
@@ -55,8 +69,222 @@ documentation for more details.
+import { ShellStep } from 'aws-cdk-lib/pipelines';
+import { GithubWorkflow } from 'cdk-pipelines-github';
+
+const app = new App();
+
+const pipeline = new GithubWorkflow(app, 'Pipeline', {
+  synth: new ShellStep('Build', {
+    commands: [
+      'yarn install',
+      'yarn build',
+    ],
+  }),
+  awsOidcRoleArn: 'arn:aws:iam::<account-id>:role/GithubActionRole',
+});
+```
+
+You can create this role either by manually setting it up, or taking advantage
+of the `AwsOidc` construct (recommended).
+
+#### `AwsOidc` Construct
+
+Because this construct involves creating an IAM role in your account, it must
+be created separate to your GitHub Workflow and deployed via a normal
+`cdk deploy` with your local AWS credentials. Upon successful deployment, the
+arn of your newly created IAM role will be exposed as a `CfnOutput`.
+
+To utilize this construct, create a separate CDK stack with the following code
+and `cdk deploy`:
+
+```ts
+import { AwsOidc } from 'cdk-pipelines-github';
+import { App, Construct, Stack, StackProps } from 'aws-cdk-lib';
+
+class MyAwsOidcRole extends Stack {
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    const provider = new AwsOidc(this, 'oidc-role', {
+      repoString: 'myUser/myRepo',
+    };
+  }
+}
+
+const app = new App();
+new MyAwsOidcRole(app, 'MyAwsOidcRole');
+app.synth();
+```
+
+Note: If you have previously created the GitHub identity provider with url
+`https://token.actions.githubusercontent.com`, the above example will fail
+because you can only have one such provider defined per account. In this
+case, you must provide the already created provider into your `AwsOidc`
+construct via the `provider` property.
+
+```ts
+class MyAwsOidcRole extends Stack {
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    const provider = new AwsOidc(this, 'oidc-role', {
+      repoString: 'myUser/myRepo',
+      provider: iam.OpenIdConnectProvider.fromOpenIdConnectProviderArn(
+        this,
+        'github',
+        'arn:aws:iam::<account-id>:oidc-provider/token.actions.githubusercontent.com',
+      ),
+    });
+  }
+}
+```
+
+#### Manual Setup
+
+In a nutshell, the IAM role you provide must reference the GitHub OIDC identity
+provider as a trusted entity. You must also set up a trust relationship between
+the IAM role and your GitHub repository. For a step-by-step tutorial on how to
+set this up, see
+[Configuring OpenID Connect in AWS](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services).
+
+In addition to setting up a relationship between GitHub and AWS, the IAM role 
+must also have permissions to assume CDK bootstrapped IAM roles and permissions 
+to access ECR repositories (if you plan on referencing Docker assets in your 
+workflow).
+
+Here is a minimum set of permissions for the IAM role: 
```

Comment:
It's great that you have this, but I would split this out into a separate doc maybe? I feel like it clutters the README a bit. At this point, 80% of the doc is on authentication, and barely anything is about the API or how you actually deploy something.

Reply:
Do you mean the manual set up, or authenticating to AWS as a whole? For now, I feel like authenticating is pretty important and deserves to be in the readme, so I moved the manual set up elsewhere.